libraries/core_libs/consensus/src/transaction/transaction_manager.cpp

```diff
@@ -349,4 +353,18 @@ bool TransactionManager::checkBlockTransactions(DagBlock const &blk) {
   return true;
 }
 
+void TransactionManager::blockFinalized(uint64_t block_number) {
+  std::unique_lock transactions_lock(transactions_mutex_);
+  auto &insufficient_balance_transactions = transactions_pool_.getInsufficientBalanceTransactions();
+  for (auto it = insufficient_balance_transactions.begin(); it != insufficient_balance_transactions.end();) {
+    if (verifyTransaction(it->second.second).first == TransactionStatus::Verified) {
+      it = insufficient_balance_transactions.erase(it);
+      transactions_pool_.insert({it->second.second, TransactionStatus::Verified});
+    } else {
+      it++;
+    }
+  }
+  transactions_pool_.blockFinalized(block_number);
```

Comment:
this should be called before for loop, otherwise we are processing txs for +1 block than specified in kTransactionsExpiryLimit

Reply:
Fixed